src/server/rpc/procedures/air-quality/air-quality.js

```diff
@@ -9,94 +9,106 @@
         city = reportingLocations[nearest.key].city,
         state = reportingLocations[nearest.key].state,
         zipcode = reportingLocations[nearest.key].zipcode;
-    logger.trace('Nearest reporting location is ' + city + ', ' + state);
+    trace('Nearest reporting location is ' + city + ', ' + state);
     return zipcode;
 };
 
-var qualityIndex = function(latitude, longitude) {
-    var response = this.response,
-        nearest,
-        url;
+/**
+ * Get air quality index of closest reporting location for coordinates
+ * @param {Latitude} latitude latitude of location
+ * @param {Longitude} longitude Longitude of location
+ * @returns {Number} AQI of closest station
+ */
+AirConsumer.qualityIndex = function(latitude, longitude) {
+    var nearest = this._getClosestReportingLocation(latitude, longitude);
 
-    logger.trace(`Requesting air quality at ${latitude}, ${longitude}`);
-    if (!latitude || !longitude) {
-        return response.status(400).send('ERROR: missing latitude or longitude');
-    }
+    trace(`Requesting air quality at ${latitude}, ${longitude} (nearest station: ${nearest})`);
 
-    nearest = getClosestReportingLocation(latitude, longitude);
-    url = baseUrl + '&zipCode=' + nearest;
-
-    logger.trace('Requesting air quality at '+ nearest);
-    
-    request(url, (err, res, body) => {
-        var aqi = -1,
-            code = err ? 500 : res.statusCode;
-
-        try {
-            body = JSON.parse(body).shift();
-            if (body && body.AQI) {
-                aqi = +body.AQI;
-                logger.trace('Air quality at '+ nearest + ' is ' + aqi);
-            }
-        } catch (e) {
-            // Just send -1 if anything bad happens
-            logger.error('Could not get air quality index: ', e);
-        }
+    return this.qualityIndexByZip(nearest);
+};
 
-        response.status(code).json(aqi);
-    });
+/**
+ * Get air quality index of closest reporting location for ZIP code
+ * @param {BoundedNumber<0,99999>} zipCode ZIP code of location
+ * @returns {Number} AQI of closest station
+ */
+AirConsumer.qualityIndexByZip = function(zipCode) {
+
+    trace(`Requesting air quality at ${zipCode}`);
 
-    return null;
+    return this._sendAnswer({queryString: `&zipCode=${zipCode}`}, '.AQI')
+    .catch(err => {
+        
+        error('Could not get air quality index: ', err);
+        
+        throw err;
+    }).then((r) => (r.length > 0? r[0]: -1));    
 };
 
-module.exports = {
-    COMPATIBILITY: {
-        path: 'air',
-        arguments: {
-            aqi: {
-                latitude: 'lat',
-                longitude: 'lng'
-            }
-        }
-    },
+/**
+ * Get air quality index of closest reporting location for coordinates
+ * @param {Latitude} latitude latitude of location
+ * @param {Longitude} longitude Longitude of location
+ * @returns {Number} AQI of closest station
+ */
+AirConsumer.aqi = function(latitude, longitude) {
```

Comment:
This method should be deprecated (add `@deprecated`)

Reply:
Fixed